src/media/remote-stream.ts

```diff
@@ -1,14 +1,87 @@
+import { AddEvents, TypedEvent, WithEventsDummyType } from '@webex/ts-events';
 import { Stream, StreamEventNames } from './stream';
 
+export enum RemoteMediaState {
+  Started = 'started',
+  Stopped = 'stopped',
+}
+
+export enum RemoteStreamEventNames {
+  MediaStateChange = 'media-state-change',
+}
+
+interface RemoteStreamEvents {
+  [RemoteStreamEventNames.MediaStateChange]: TypedEvent<(state: RemoteMediaState) => void>;
+}
+
 /**
  * A stream originating from a remote peer.
  */
-export class RemoteStream extends Stream {
+class _RemoteStream extends Stream {
+  [RemoteStreamEventNames.MediaStateChange] = new TypedEvent<(state: RemoteMediaState) => void>();
+
+  /**
+   * Create a RemoteStream from the given values.
+   *
+   * @param stream - The initial output MediaStream for this Stream.
+   */
+  constructor(stream: MediaStream) {
+    super(stream);
+    this.handleMediaStarted = this.handleMediaStarted.bind(this);
+    this.handleMediaStopped = this.handleMediaStopped.bind(this);
+    this.outputTrack.addEventListener('mute', this.handleMediaStopped);
+    this.outputTrack.addEventListener('unmute', this.handleMediaStarted);
+  }
+
   /**
    * @inheritdoc
    */
-  get muted(): boolean {
-    return !this.outputTrack.enabled;
+  protected handleMediaStarted(): void {
+    this[RemoteStreamEventNames.MediaStateChange].emit(RemoteMediaState.Started);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  protected handleMediaStopped(): void {
+    this[RemoteStreamEventNames.MediaStateChange].emit(RemoteMediaState.Stopped);
+  }
+
+  /**
+   * Helper function to add event handlers to a MediaStreamTrack. See
+   * {@link Stream.addTrackHandlersForStreamEvents} for why this is useful.
+   *
+   * @param track - The MediaStreamTrack.
+   */
+  private addTrackHandlersForRemoteStreamEvents(track: MediaStreamTrack): void {
+    track.addEventListener('mute', this.handleMediaStopped);
+    track.addEventListener('unmute', this.handleMediaStarted);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  protected addTrackHandlers(track: MediaStreamTrack): void {
```

Comment:
what happens when `RemoteStream.replaceTrack()` is called and the previous track was muted, but the new one isn't. Should we emit the MediaStateChange event?

Reply:
There's no "mute" for remote (something we're trying to correct with new vocabulary), but I think this point still stands in that the media states could differ...maybe we need to compare the old and new states at replacement time.

Reply:
That's a good point. Yeah, I think we should.